packages/api/src/modules/providers/subventia/adapters/subventia.adapter.test.ts

```diff
@@ -1,35 +1,102 @@
         service_instructeur: "CIPDR",
         annee_demande: 2023,
         siret: "123456789",
-        date_commision: "12/07/2023",
+        date_commision: new Date("2023-04-01T00:00:00.000Z"),
         montants_accorde: 100,
         montants_demande: 600,
         dispositif: "FIPDR",
         sous_dispositif: "",
         status: "Refused",
-        provider: "subventia",
+        statut_label: ApplicationStatus.REFUSED,
     };
 
+    const exportDate = new Date("2022-08-02T00:00:00.000Z");
+    const entity = { ...entityIncomplete, provider: "subventia", exportDate: exportDate };
+
+    const dbo = { ...entity, __data__: [], _id: new ObjectId("60d5ec9af682fbd39a1b57a2") };
+
     describe("applicationToEntity", () => {
         it("should call indexDataByPathObject", () => {
-            let mockIndexDataByPathObject = jest.spyOn(ParseHelper, "indexDataByPathObject").mockReturnValue(entity);
+            let mockIndexDataByPathObject = jest
+                .spyOn(ParseHelper, "indexDataByPathObject")
+                .mockReturnValue(entityIncomplete);
 
-            SubventiaAdapter.applicationToEntity(application);
+            SubventiaAdapter.applicationToEntity(application, exportDate);
             expect(mockIndexDataByPathObject).toHaveBeenCalled();
         });
 
         it("should return entity", () => {
             const expected = entity;
-            const actual = SubventiaAdapter.applicationToEntity(application);
+            const actual = SubventiaAdapter.applicationToEntity(application, exportDate);
+            expect(actual).toEqual(expected);
+        });
+    });
+
+    describe("toDemandeSubventionDto", () => {
+        let mockToPV = jest.fn().mockImplementation(value => value);
+        let mockBuildProviderValueAdapter = jest
+            .spyOn(ProviderValueFactory, "buildProviderValueAdapter")
+            .mockReturnValue(mockToPV);
+
+        it("should call ProviderValueFactory.buildProviderValueAdapter once", () => {
+            SubventiaAdapter.toDemandeSubventionDto(dbo);
+            expect(mockBuildProviderValueAdapter).toHaveBeenCalledWith("Subventia", exportDate);
+        });
+
+        it("should call toPV with ten different attributs", () => {
+            const attributes = [
+                "siret",
+                "service_instructeur",
+                "status",
+                "statut_label",
+                "montants_accorde",
+                "montants_demande",
+                "date_commision",
+                "annee_demande",
+                "dispositif",
+                "sous_dispositif",
+            ];
+
+            SubventiaAdapter.toDemandeSubventionDto(dbo);
+            attributes.forEach(attr => {
+                expect(mockToPV).toHaveBeenCalledWith(dbo[attr]);
+            });
+        });
+
+        it("should return DemandeSubvention", () => {
+            mockBuildProviderValueAdapter.mockRestore();
+            mockToPV.mockRestore();
```

Comment:
pourquoi tu commences par `mockRestore` des trucs ? En principe s'il faut restore c'est soit dans un `afterEach`, soit dans un `afterAll`, soit à la *fin* d'un test individuel

Reply:
parce que je voulais le restore que pour cette it() là et je ne vois pas trop pourquoi ça ne se fait pas :) Il vaut mieux le restore à la fin du it précedent ? Je me disais que comme des fois je le mock d'une façon particulière pour un test it donné. Là je pouvais faire un mockrestore au début du test. Je vais bien que on échange là dessus pour mieux comprendre. Merci!

Reply:
restore c'est "je veux tout effacer et remplacer par l'implémentation originale" donc en général on le fait à l'intérieur du `describe` qui mock la méthode.

C'est quasi exclusivement utilisé pour le seul cas ou l'on mock une méthode du fichier tester.

Par exemple dans le test de `subventia.adapter` si une de ses méthodes appelle une autre méthode interne, on peut vouloir mock la seconde méthode. Mais pour pouvoir tester unitairement cette seconde méthode on a besoin que le mock soit restoré. 

Deux choix s'offre à nous :
1. C'est le `describe` qui mock la méthode qui s'occupe de `mockRestore`  à la fin
2. C'est le `describe` de la méthode concernée qui s'occupe de restaurer sa propre définition (peut-être envisageable si la méthode est mock pour l'ensemble du fichier

Je crois que de manière général c'est plutôt le 1. qui est utilisée 

Par contre on ne restaure pas de méthode autre que celle qu'on teste au début d'un `describe`. Il n'y a aucun cas ou on veut restaurer une autre méthode que celle testée, sinon ce n'est pas un test unitaire. Ou alors tu voulais utiliser `mockReset` ! :)

Si c'est pas clair on peut en parler de vive voix